### PR TITLE
Feat/m overlay subnet

### PR DESCRIPTION
**Pull request by Abhishekism9450**
Added:-

- is_external attribute for External subnet
- enable_nat
- vpc_reference_uuid to support Overlay subnet

### CODE REVIEW

File: examples/external_subnets/main.tf [hunk 1]
```diff
@@ -0,0 +1,74 @@
+terraform {
+  required_providers {
+    nutanix = {
+      source  = "nutanixtemp/nutanix"
```
Comment (bhati-pradeep) [inline examples/external_subnets/main.tf]: Change source and version before landing

File: examples/external_subnets/main.tf [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+resource "nutanix_subnet" "accNat" {
+    # General Info
+	name        = "test-ext-sub-with-nat"
+	description = "Description of my test VLAN updated"
+
+    # subnet_type should be VLAN for external subnet with NAT
+	subnet_type = "VLAN"
+    cluster_uuid = local.cluster1
+    vlan_id = 121
+	subnet_ip          = "10.xx.xx.xx"
```
Comment (bhati-pradeep) [inline examples/external_subnets/main.tf]: check formatting

File: examples/overlay_subnets/main.tf [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+	cluster.metadata.uuid if cluster.service_list[0] != "PRISM_CENTRAL"
+	][0]
+}
+  
+
+# overlay subnet resource
+
+resource "nutanix_subnet" "acc" {
+    # General Information
+	name        = "test-overlay-subnet"
```
Comment (bhati-pradeep) [inline examples/overlay_subnets/main.tf]: check formatting

File: examples/overlay_subnets/terraform.tfvars [hunk 4]
```diff
@@ -0,0 +1,5 @@
+#define values to the variables to be used in terraform file
+nutanix_username = "admin"
+nutanix_password = "password"
+nutanix_endpoint = "10.xx.xx.xx"
+vpc_reference_uuid = "string"
```
Comment (bhati-pradeep) [inline examples/overlay_subnets/terraform.tfvars]:
```suggestion
vpc_reference_uuid = "<vpc-uuid>"
```

File: nutanix/data_source_nutanix_vpc_list_test.go [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+	common_domain_name_server_ip_list{
+			ip = "8.8.8.9"
+	}
+  
+	externally_routable_prefix_list{
+	  ip=  "172.31.0.0"
+	  prefix_length= 16
+	}
+  }
+	data "nutanix_vpc_list" "test" {
```
Comment (bhati-pradeep) [inline nutanix/data_source_nutanix_vpc_list_test.go:96]: test is about querying with name but no vpc name mentioned for query. If covered in nutanix_vpc data source test then combine this with first test in this file as it would also require some VPCs to be present initially before
datasource can call

File: nutanix/data_source_nutanix_vpc_list_test.go [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+
+func TestAccNutanixVPCListDataSource_basic(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccVPCListDataSourceConfig(),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrSet("data.nutanix_vpc_list.test", "entities.#"),
```
Comment (bhati-pradeep) [inline nutanix/data_source_nutanix_vpc_list_test.go:18]: Also check if entities is not empty

File: nutanix/resource_nutanix_subnet.go [hunk 7]
```diff
@@ -671,15 +717,28 @@ func getSubnetResources(d *schema.ResourceData, subnet *v3.SubnetResources) {
 		dhcpo.DomainSearchList = expandStringList(v.([]interface{}))
 	}
 
-	v, ok := d.GetOk("vlan_id")
-	if v.(int) == 0 || ok {
-		subnet.VlanID = utils.Int64Ptr(int64(v.(int)))
+	if v, ok := d.GetOk("vlan_id"); ok {
```
Comment (bhati-pradeep) [inline nutanix/resource_nutanix_subnet.go:720]: why this change is required ?

Comment (Abhishekism9450) [reply nutanix/resource_nutanix_subnet.go:720]: this change is required as upcoming overlay subnets will not require vlan id . it requires vpc reference .

File: nutanix/data_source_nutanix_vpc_list_test.go (hunk 6 above)
Comment (Abhishekism9450) [reply nutanix/data_source_nutanix_vpc_list_test.go:18]: entities can be empty if no vpc is present.

**Comment by abhimutant** [comment]
/ok-to-test pc
>Acceptance test run status: cancelled
>Current line coverage (0.0) is below threshold (50). Kindly add more acceptance tests.

**Comment by abhimutant** [comment]
/ok-to-test pc
>Acceptance test run status: failure
>Current line coverage (42.0) is below threshold (50). Kindly add more acceptance tests.

**Review by siddharth-kulshrestha — APPROVED** [review]
